auth/src/main/java/com/firebase/ui/auth/util/SmartLock.java

```diff
@@ -252,12 +255,18 @@ public void saveCredentialsOrFinish(AppCompatBase activity,
             return;
         }
 
-        mCredentialsApiClient = new GoogleApiClient.Builder(activity)
+        if (mActivity.isFinishing()) {
+            finish();
+            return;
+        }
+
+        mCredentialsApiClient = new Builder(activity)
                 .addConnectionCallbacks(this)
                 .addOnConnectionFailedListener(this)
                 .addApi(Auth.CREDENTIALS_API)
                 .enableAutoManage(activity, this)
                 .build();
+        mCredentialsApiClient.connect();
```

Comment:
Nothing wrong with this but the fact that you had to do it is concerning.  If you can reliably reproduce a situation where this is critical could you file a bug and tell the Gms client folks about it?